src/main/java/org/apache/maven/shared/jar/classes/JarClassesAnalysis.java

```diff
@@ -51,8 +57,17 @@
 @Named
 @SuppressWarnings("checkstyle:MagicNumber")
 public class JarClassesAnalysis {
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
+    /**
+     * Constant representing the classes in the root of a Multi-Release JAR file.
+     * Meaning outside of any given META-INF/versions/NN/... entry.
+     */
+    private static final Integer ROOT = 0;
+
+    private static final Pattern ENTRY_FILTER_MULTI_RELEASE = Pattern.compile("^META-INF/versions/(\\d{1,2})/.*$");
```

Comment:
Rather `.+`? Empty entry is pointless no? Or should it also apply to the dir entry?

Reply:
I don't foresee any use for storing an empty entry. Should the user want to just query if an specific version is present, it can use other means, like querying the JarRuntimeVersions.getRuntimeVersionMap().

Some corner cases to think: which version should a jar with just a single 'module-info.class' file should return? How about a JAR with a single file 'META-INF/versions/17/module-info.class'? it is 17? or it is whatever the module-info. class target version is?

Reply:
I don't that that only having `module-info.class` is unsual, there might be cases you need that info to make it properly work for modules, but simply don't need for 8. So for me empty does not qualify, at least one class should be in...